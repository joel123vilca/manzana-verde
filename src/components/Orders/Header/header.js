import { mapState } from "vuex";
export default {
  data() {
    return {
      value1: 50
    };
  },
  computed: {
    ...mapState({
      calories: state => state.calories
    })
  },
  methods: {
    format() {
      return 1100;
    }
  }
};
