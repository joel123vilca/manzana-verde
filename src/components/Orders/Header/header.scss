.orders__header {
  height: 100px;
  border-bottom: 2px solid #e8e8e8;
  .header__profile {
    display: flex;
    justify-content: space-between;
    .profile__avatar {
      width: 30px;
      height: 30px;
      border-radius: 50%;
      border: 1px solid black;
      img {
        width: 100%;
        border-radius: 50%;
        object-fit: contain;
      }
    }
    .profile__title {
      size: 20px;
      color: #606060;
      font-style: normal;
      font-weight: 700;
      display: flex;
      align-items: center;
      justify-content: center;
      padding-left: 10px;
      padding-right: 10px;
    }
    .profile__count {
      width: 40px;
      height: 40px;
    }
    .profile__status {
      width: 85px;
      height: 30px;
      margin-left: 10px;
    }
    .profile__notification {
      width: 50px;
      height: 30px;
      margin-left: 10px;
    }
  }
  .header__info {
    display: flex;
    padding: 10px 0 0 10px;
    justify-content: space-between;
  }
  .progress {
    padding-top: 0px;
  }
}
