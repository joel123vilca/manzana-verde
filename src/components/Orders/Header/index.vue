<template>
  <div class="orders__header">
    <div class="header__profile">
      <div class="profile__avatar">
        <img :src="require('@/assets/logo.png')" />
      </div>
      <div class="profile__title">Mis Pedidos</div>
      <div class="profile__count">
        <el-progress
          type="circle"
          :percentage="50"
          :width="30"
          :show-text="false"
          color="#35B266"
        ></el-progress>
      </div>
      <div class="profile__status">
        <img :src="require('@/assets/status.svg')" />
      </div>
      <div class="profile__notification">
        <img :src="require('@/assets/notification.svg')" />
      </div>
    </div>
    <div class="header__info">
      <span>0kal</span>
      <span>22000kcal</span>
    </div>
    <el-progress
      :percentage="calories"
      color="#FFD538"
      :format="format"
      class="progress"
    ></el-progress>
  </div>
</template>
<script src="./header.js"></script>
<style src="./header.scss" scoped lang="scss"></style>
