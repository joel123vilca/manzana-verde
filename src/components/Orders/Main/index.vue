<template>
  <div class="home">
    <div class="dates">
      <div
        v-for="(date, index) in dates"
        :key="date.num"
        class="date"
        :class="{ active: date.active }"
        @click="selectDate(index)"
      >
        <span>{{ date.title }} </span>
        <div class="number" :class="{ select: date.select }">
          <h3>{{ date.num }}</h3>
        </div>
      </div>
    </div>
    <div class="address">
      <i class="el-icon-s-cooperation" style="color:#35B266; font-size: 40px"></i>
      <div>
        <span>Entregar en:</span>
        <p>Av. Mariscal Ramon Castilla</p>
      </div>
    </div>
    <div v-for="(item, index) in listMain" :key="index">
      <div
        v-if="item.foodsAsign.length === 0"
        class="home__lunch"
        :class="{ desactive: !item.active }"
      >
        <div class="lunch__icon">
          <img :src="require(`@/assets/${item.icon}`)" alt="lunch icon" />
        </div>
        <div class="lunch__description">
          <span :class="{ title: !item.active }">{{ item.name }} </span>
          <p>{{ item.value }}</p>
        </div>
        <div class="lunch__action">
          <button
            v-if="item.active"
            class="add"
            @click="$router.push({ name: 'About', params: { slug: item.slug } })"
          >
            Agregar
          </button>
          <span v-else>No tienes creditos</span>
        </div>
      </div>
      <div v-else class="food__list" :class="{ desactive: !item.active }">
        <div v-for="food in item.foodsAsign" :key="food.id" class="food__item">
          <div class="food__image">
            <img :src="require('@/assets/food.png')" alt="food image" />
          </div>
          <div
            class="food__description"
            @click="$router.push({ name: 'Detail', params: { id: food.id } })"
          >
            <span>{{ food.name }} </span>
            <p>{{ item.value }}</p>
          </div>
          <div class="food__action">
            <i
              class="el-icon-delete"
              style="color:#FF9797"
              @click="filterFood(item.id, food.id)"
            ></i>
          </div>
        </div>
      </div>
    </div>
  </div>
</template>
<script src="./list.js"></script>
<style src="./main.scss" scoped lang="scss"></style>
