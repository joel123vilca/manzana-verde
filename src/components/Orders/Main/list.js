import { mapState, mapActions } from "vuex";
export default {
  data() {
    return {
      isActive: true,
      dates: [
        {
          title: "Lun",
          num: 29,
          select: false,
          active: true
        },
        {
          title: "Mar",
          num: 30,
          select: true,
          active: false
        },
        {
          title: "Mie",
          num: "01",
          select: false,
          active: false
        },
        {
          title: "Jue",
          num: "02",
          select: false,
          active: false
        },
        {
          title: "Vie",
          num: "03",
          select: false,
          active: false
        }
      ]
    };
  },
  computed: {
    ...mapState({
      listMain: state => state.listMain
    })
  },
  methods: {
    ...mapActions({
      deleteFood: "DELETE_FOOD"
    }),
    filterFood(id, foodId) {
      this.deleteFood({ id, foodId });
    },
    selectDate(index) {
      this.dates = this.dates.map((item, i) => {
        if (i == index) {
          item.active = true;
        } else {
          item.active = false;
        }
        return item;
      });
    }
  }
};
