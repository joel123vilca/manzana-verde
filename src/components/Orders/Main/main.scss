.home {
  padding: 10px;
  .dates {
    display: flex;
    align-items: center;
    justify-content: space-between;
    border-bottom: 2px solid #e8e8e8;
    padding-bottom: 5px;
    margin-bottom: 5px;
    .date {
      width: 60px;
      height: 55px;
      border-radius: 10px;
      background-color: white;
      padding-top: 5px;
      display: flex;
      align-items: center;
      flex-direction: column;
      color: #c9c9c9;
      cursor: pointer;
      .number {
        height: 25px;
        width: 25px;
        h3 {
          margin-top: 0px;
        }
      }
      .select {
        border-bottom: 2px solid #35b266;
      }
    }
    .active {
      background-color: #ffd538;
      color: #606060;
    }
  }
  .address {
    display: flex;
    align-content: center;
    border-bottom: 2px solid #e8e8e8;
    div {
      display: flex;
      flex-direction: column;
      align-items: flex-start;
      padding-left: 10px;
    }
    span {
      font-weight: 700;
    }
    p {
      margin-top: 0px;
    }
  }
  .home__lunch {
    margin-top: 10px;
    border: 2px solid #e8e8e8;
    border-radius: 10px;
    height: 90px;
    display: flex;
    align-items: center;
    .lunch__icon {
      width: 40px;
      height: 40px;
      margin-left: 10px;
    }
    .lunch__description {
      width: 140px;
      height: 30px;
      margin-left: 5px;
      span {
        color: #35b266;
        size: 12px;
      }
      .title {
        color: #606060;
      }
      p {
        margin-top: 0px;
        color: #a9a9a9;
        size: 10px;
      }
    }
    .lunch__action {
      width: 180px;
      margin-left: 30px;
      .add {
        width: 130px;
        height: 40px;
        size: 15px;
        border: 1px solid #ffd538;
        background: #ffd538;
        color: #606060;
        border-radius: 10px;
      }
      span {
        margin-top: 10px;
        color: #a9a9a9;
        font-size: 15px;
      }
    }
  }
  .desactive {
    background-color: #fafafa;
  }
  .food__list {
    margin-top: 10px;
    border: 2px solid #e8e8e8;
    border-radius: 10px;
    .food__item:first-child {
      border-top: 0px solid #e8e8e8;
    }
    .food__item {
      height: 90px;
      display: flex;
      margin: 0 10px 0 10px;
      justify-content: space-between;
      align-items: center;
      border-top: 2px solid #e8e8e8;
      .food__image {
        margin-top: 10px;
        img {
          border-radius: 5px;
          width: 70px;
          height: 60px;
          object-fit: cover;
        }
      }
      .food__description {
        margin-top: 10px;
        display: flex;
        flex-direction: column;
        align-items: flex-start;
        span {
          color: #35b266;
          size: 12px;
        }
        p {
          margin-top: 0px;
          color: #a9a9a9;
          size: 10px;
        }
      }
      .food__action {
        cursor: pointer;
      }
    }
  }
  .dialog__image {
    width: 100%;
  }
}

@media (min-width: 992px) {
  .home {
    display: grid;
    grid-template-columns: 400px 400px;
    grid-gap: 50px;
  }
}
