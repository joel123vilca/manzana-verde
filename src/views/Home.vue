<template>
  <div>
    <Header />
    <Main />
  </div>
</template>

<script>
import Header from "@/components/Orders/Header";
import Main from "@/components/Orders/Main";

export default {
  name: "Home",
  components: {
    Header,
    Main
  }
};
</script>
