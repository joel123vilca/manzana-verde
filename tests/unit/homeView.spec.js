import { shallowMount } from "@vue/test-utils";
import Home from "@/views/Home";
import Header from "@/components/Orders/Header";
import Main from "@/components/Orders/Main";

describe("home view", () => {
  const build = () => {
    const wrapper = shallowMount(Home);

    return {
      wrapper,
      Header: () => wrapper.find(Header),
      Main: () => wrapper.find(Main)
    };
  };

  it("renders the component", () => {
    const { wrapper } = build();
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("renders the right", () => {
    const { Header, Main } = build();
    expect(Header().exists()).toBe(true);
    expect(Main().exists()).toBe(true);
  });
});
