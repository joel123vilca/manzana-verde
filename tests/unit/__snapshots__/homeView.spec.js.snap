// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home view renders the component 1`] = `
<div>
  <header-stub></header-stub>
  <main-stub></main-stub>
</div>
`;
